src/Swarm/TUI/Model/Repl.hs

```diff
@@ -185,6 +186,11 @@ getLatestREPLHistoryItems n h = toList latestN
   latestN = Seq.drop oldestIndex $ h ^. replSeq
   oldestIndex = max (h ^. replStart) $ length (h ^. replSeq) - n
 
+-- | Get only the items from the REPL history that were entered during
+--   the current session.
+getSessionREPLHistoryItems :: REPLHistory -> [REPLHistItem]
+getSessionREPLHistoryItems h = toList $ Seq.drop (h ^. replStart) (h ^. replSeq)
```

Comment:
@byorgey honestly I might be asking you for a solution to my 10000 messages issue, so feel free to leave it as is. 😅

Reply:
I think there's definitely some things we can do to make it more efficient, converting the entire history to a list and having brick render the whole thing is probably not a good idea.  I will look into it.

Reply:
Ah, unfortunately using brick's `GenericList` rendering machinery will not work, since it assumes that every list item has the same height.  Cutting the `Seq` down to a smaller size before handing it off to `brick` to render will definitely save time, but to make that work I guess we would have to manually tell `brick` how big to make the scrollbar.  I don't know if that is possible.

Reply:
I think I found a good solution re: efficiency, which is to cache the REPL history so it is only re-rendered every time a new input is entered or a new output becomes available.